src/Shared/UnitTests/TypeLoader_Tests.cs

```diff
@@ -90,6 +91,30 @@ public void LoadInsideAsssembly()
             }
         }
 
+        [Fact]
+        public void LoadTaskDependingOnMSBuild()
+        {
+            using (TestEnvironment env = TestEnvironment.Create())
+            {
+                TransientTestFolder folder = env.CreateFolder(createFolder: true);
+                string currentAssembly = Assembly.GetExecutingAssembly().Location;
+                string utilitiesName = "Microsoft.Build.Utilities.Core.dll";
+                string newAssemblyLocation = Path.Combine(folder.Path, Path.GetFileName(currentAssembly));
+                string portableTaskFolderPath = Path.GetFullPath(
+                    Path.Combine(BuildEnvironmentHelper.Instance.CurrentMSBuildToolsDirectory, "..", "..", "..", "Samples", "PortableTask"));
+
+                // The "first" directory is "Debug" or "Release"
+                portableTaskFolderPath = Path.Combine(Directory.GetDirectories(portableTaskFolderPath).First(), "netstandard2.0", "OldMSBuild");
```

Comment:
I do not like this coupling at all. Can you not copy `Utilities.Core.dll` in the build of this project, so it's in a known relative location?

Reply:
This is how it's done in PortableTasks_Tests:
https://github.com/dotnet/msbuild/blob/main/src/Tasks.UnitTests/PortableTasks_Tests.cs#L21
https://github.com/dotnet/msbuild/blob/main/src/Tasks.UnitTests/PortableTasks_Tests.cs#L57

I think it's reasonable to copy what we've previously done.

Reply:
Well that's horrifying. It also uses a static field--can you simplify so it only exists once, at least?